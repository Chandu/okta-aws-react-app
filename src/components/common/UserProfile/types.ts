export interface Props {
    showSeparator?: boolean;
}
