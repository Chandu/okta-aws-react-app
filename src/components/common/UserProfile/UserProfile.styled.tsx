import { Popover } from '@medly-components/core';
import { css, styled } from '@utils';

const avatarSeparator = css`
    &::before {
        content: '';
        position: absolute;
        left: -1.2rem;
        width: 1px;
        height: 3.2rem;
        background: ${({ theme }) => theme.colors.grey[200]};
        margin: auto;
        top: 0;
        bottom: 0;
    }
`;

export const UserProfilePopover = styled(Popover)<{ leftMargin?: string; showSeparator: boolean }>`
    && {
        margin-left: ${({ leftMargin }) => (leftMargin ? leftMargin : '2.4rem')};
    }
    ${props => props.showSeparator && avatarSeparator};
`;
