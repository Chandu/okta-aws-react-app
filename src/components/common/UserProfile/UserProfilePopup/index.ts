import { UserProfilePopup } from './UserProfilePopup.component';

export default UserProfilePopup;
