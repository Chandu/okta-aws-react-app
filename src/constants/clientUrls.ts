export const URLRoot = {
    dashboard: '/',
    dataUpload: '/data-upload'
};

export const AuthURL = {
    login: '/login'
};
