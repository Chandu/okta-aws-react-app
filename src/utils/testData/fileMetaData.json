{
    "description": "dummy description",
    "uploaded-by-user-name": "xyz",
    "uploaded-timestamp": "Fri Apr 29 2022 18:46:10 GMT+0530 (India Standard Time)",
    "uploaded-user-email": "xyz@gmail.com"
}
