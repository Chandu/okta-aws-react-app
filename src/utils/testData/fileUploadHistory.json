[
    {
        "key": "1651756594298_dummyFile.csv",
        "description": "dummy description",
        "fileName": "1651756594298_dummyFile.csv",
        "uploaded-by-user-name": "xyz",
        "uploaded-timestamp": "Thu May 05 2022 19:28:07 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile1.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile1.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile3.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile3.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile4.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile4.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile5.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile5.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile6.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile6.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile7.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile7.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile8.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile8.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile9.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile9.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile10.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile10.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    },
    {
        "key": "1651756594296_dummyFile11.csv",
        "description": "dummy description",
        "fileName": "1651756594296_dummyFile11.csv",
        "uploaded-by-user-name": "xyz1",
        "uploaded-timestamp": "Thu May 05 2022 18:46:34 GMT+0530 (India Standard Time)",
        "uploaded-user-email": "xyz1@tango340b.com"
    }
]
