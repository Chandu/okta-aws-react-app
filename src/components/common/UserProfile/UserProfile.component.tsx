import { useOktaAuth } from '@okta/okta-react';
import React, { useCallback, useEffect, useState } from 'react';
import { Props } from './types';
import { UserProfilePopover } from './UserProfile.styled';
import { UserProfilePopup } from './UserProfilePopup/UserProfilePopup.component';

export const UserProfile: React.FunctionComponent<Props> = React.memo(({ showSeparator = false }) => {
    const { authState, oktaAuth } = useOktaAuth(),
        [userInfo, setUserInfo] = useState<any>({}),
        handleSignOut = useCallback(() => oktaAuth.signOut?.(), [oktaAuth]);
    useEffect(() => {
        if (!authState || !authState.isAuthenticated) setUserInfo({});
        else setUserInfo(authState?.idToken?.claims || {});
    }, [authState, oktaAuth]);

    return (
        <UserProfilePopover interactionType="click" showSeparator={showSeparator}>
            <UserProfilePopup signOut={handleSignOut} userName={userInfo?.name || ''} showSeparator={showSeparator} />
        </UserProfilePopover>
    );
});
UserProfile.displayName = 'UserProfile';
UserProfile.defaultProps = {
    showSeparator: false
};
