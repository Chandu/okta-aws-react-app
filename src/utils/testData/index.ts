import fileMetaData from './fileMetaData.json';
import fileUploadHistory from './fileUploadHistory.json';
import oktaStorage from './oktaStorage.json';

export { oktaStorage, fileMetaData, fileUploadHistory };
