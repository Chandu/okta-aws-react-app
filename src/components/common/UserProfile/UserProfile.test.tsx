import { URLRoot } from '@constants/clientUrls';
import { cleanup, fireEvent, renderWithRouter, screen } from '@test-utils';
import { oktaStorage } from '@utils/testData';
import { MemoryRouter } from 'react-router-dom';
import { UserProfile } from './UserProfile.component';

const mockSignOut = jest.fn(),
    mockSignInWithRedirect = jest.fn(),
    mockSignInWithCredentials = jest.fn(),
    mockIsAuthenticated = jest.fn(() => Promise.resolve({ data: true }));

const renderUserProfile = () =>
    renderWithRouter(
        <MemoryRouter initialEntries={['/']}>
            <UserProfile />
        </MemoryRouter>
    );

const mockHistoryPush = jest.fn();
const mockDashboardUrl = URLRoot.dashboard;

jest.mock('react-router-dom', () => ({
    ...(jest.requireActual('react-router-dom') as any),
    useHistory: () => ({
        push: mockHistoryPush
    }),
    useLocation: () => ({
        pathname: mockDashboardUrl
    })
}));

jest.mock('@okta/okta-react', () => {
    return {
        useOktaAuth: () => ({
            oktaAuth: {
                signOut: mockSignOut
            },
            authState: {}
        })
    };
});

describe('UserProfile', () => {
    beforeAll(() => {
        localStorage.setItem('okta-token-storage', JSON.stringify(oktaStorage));
    });

    afterEach(() => {
        cleanup();
        jest.clearAllMocks();
    });

    it('should render properly', () => {
        const { container } = renderUserProfile();
        expect(container).toMatchSnapshot();
    });

    it('should use username from local storage', () => {
        renderUserProfile();
        const userInitials = screen.getByText('DN');
        expect(userInitials).toBeInTheDocument();
        fireEvent.click(userInitials);
        const userName = screen.getByText('dummy name');
        expect(userName).toBeInTheDocument();
        expect(userName).toHaveStyle({
            textTransform: 'uppercase'
        });
    });

    it('should call signOut fn', () => {
        renderUserProfile();
        fireEvent.click(screen.getByText('DN'));
        fireEvent.click(screen.getByText('Sign Out'));
        expect(mockSignOut).toBeCalled();
    });
});
