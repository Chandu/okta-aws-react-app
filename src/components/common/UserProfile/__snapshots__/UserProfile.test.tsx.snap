// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should render properly 1`] = `
.c5 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2rem;
  text-align: initial;
  text-transform: uppercase;
  display: inline-block;
}

.c2 {
  display: inline-block;
  text-align: center;
  min-width: -webkit-max-content;
  min-width: -moz-max-content;
  min-width: max-content;
  width: 4rem;
  height: 4rem;
  border-radius: 0.4rem;
  overflow: hidden;
  cursor: inherit;
  color: #6E14FF;
  background: #E9DCFF;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 .c4 {
  line-height: 4rem;
  font-size: 1.6rem;
  font-weight: 400;
  font-family: Poppins;
}

.c2 img {
  width: 4rem;
  height: 4rem;
  object-fit: cover;
  border: 0.1rem solid #dfe4e9;
  box-sizing: border-box;
  border-radius: 0.4rem;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1.c1 {
  margin-left: 2.4rem;
}

.c3 {
  z-index: 12;
  position: relative;
  cursor: pointer;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  overflow: visible;
}

<div>
  <div
    class="c0 c1"
    id="medly-popover-wrapper"
  >
    <div
      class="c2 c3"
    >
      <strong
        class="c4 c5"
      >
        DN
      </strong>
    </div>
  </div>
</div>
`;
